File: 26-07-2025_Majority Element - More Than ndivide3.java


class Solution {
    public List<Integer> findMajority(int[] arr) {
        int value = arr.length / 3;
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int num : arr) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        
        List<Integer> res = new ArrayList<>();
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() > value) {
                res.add(entry.getKey());
            }
        }
         Collections.sort(res);
        return res;
    }
}

File: 26-07-2025_Majority Element - More Than ndivide3.py
class Solution:
    def findMajority(self, arr):
        value = len(arr) // 3
        count_map = {}
        
        for num in arr:
            count_map[num] = count_map.get(num, 0) + 1
            
        res = []
        for key, count in count_map.items():
            if count > value:
                res.append(key)
        res.sort()        
        return res
